frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx

```diff
@@ -18,14 +22,52 @@ export const getKnativeSidepanelSinkSection = (element: GraphElement) => {
   }
   if (
     isDynamicEventResourceKind(referenceFor(resource)) ||
-    (resource.kind === CamelKameletBindingModel.kind && data.kameletType === KameletType.Source)
+    (resource.kind === CamelKameletBindingModel.kind &&
+      data.data.kameletType === KameletType.Source)
   ) {
     return (
       <TopologySideBarTabSection>
-        <EventSourceResources
-          obj={resource}
-          ownedSources={element.getData().resources.eventSources}
-        />
+        <EventSourceTarget obj={resource} />
+      </TopologySideBarTabSection>
+    );
+  }
+  return undefined;
+};
+
+export const getKnativeSidepanelSinkAssociatedDeployment = (element: GraphElement) => {
+  const resource = getResource(element);
+  const data = element.getData();
+  if (!resource) {
+    return undefined;
+  }
+  if (
+    isDynamicEventResourceKind(referenceFor(resource)) ||
+    (resource.kind === CamelKameletBindingModel.kind &&
+      data.data.kameletType === KameletType.Source)
+  ) {
+    return (
+      <TopologySideBarTabSection>
+        <EventSourceDeployments deploymentObj={element.getData().resources?.associatedDeployment} />
+      </TopologySideBarTabSection>
+    );
+  }
+  return undefined;
+};
+
+export const getKnativeSidepanelSinkEventSources = (element: GraphElement) => {
+  const resource = getResource(element);
+  const data = element.getData();
+  if (!resource) {
+    return undefined;
+  }
```

Comment:
we can do early return if resources also don't exists and any reason to return `undefined` over `null` ?

```suggestion
  if (!resource || !data?.resources) {
    return null;
  }
```